### PR TITLE
doc: Add some documentation about process and directory structure

### PR DESCRIPTION
**Pull request by laanwj**

### CODE REVIEW

**Comment by Emzy** [comment]
ACK 01cd0aa

File: README.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
 This repository contains Guix attestations for releases of Bitcoin Core.
 
 See https://github.com/bitcoin/bitcoin/blob/master/doc/release-process.md#building on how to build the release with Guix and create an attestation.
+
+## Overall process
+
+The Guix build consists of two stages:
+
+- In the first stage (`noncodesigned`), people compile the binaries from source.
+- Then, code signatures for Windows and MacOS are generated from the binaries that were produced in the first stage, and distributed to the builders.
```
Comment (TheCharlatan) [inline README.md:10]: Maybe link to the detached sigs repo https://github.com/bitcoin-core/bitcoin-detached-sigs ?

Comment (laanwj) [reply README.md:10]: Sure, happy to add that, though this is really only meant as a high-level overview not detailed instructions.

Comment (laanwj) [reply README.md:10]: Thinking of it, it might make sense to move that line *under* here.

**Review by glozow — COMMENTED** [review]
Nice! text lgtm

File: README.md (hunk 1 above)
Comment (laanwj) [reply README.md]: I've added a link (made the word "distributed" a link to that repo), and moved the detailed process link to below the overall process description.

**Review by TheCharlatan — APPROVED** [review]
ACK 4e25c8c1078d45bd376654f23de9d1ea1c726c97

**Review by guggero — APPROVED** [review]
ACK 4e25c8c1078d45bd376654f23de9d1ea1c726c97

**Review by RanaAli996 — APPROVED** [review]

**Pull request merged**